lib/Bio/EnsEMBL/DataCheck/Test/Compara.pm

```diff
@@ -145,4 +148,39 @@ sub cmp_tag {
   return $result;
 }
 
+=item B<check_id_range>
+
+check_id_range($dba, $table_name, $id_column, $id_column_val);
+
+Takes C<$table_name> and $table_name . "_id" in C<$table_name> and checks that
+$table_name . "_id" is offset by C<$id_column> using C<$id_column_val>.
+
+This is an offset type commonly used in compara databases using genome_db_id or
+method_link_species_set_id for example.
+
+=back
+
+=cut
+
+sub check_id_range {
+  my ($dba, $table_name, $id_column, $id_column_val) = @_;
+
+  my $helper = $dba->dbc->sql_helper;
+  # The naming convention of the id column in table_name follows table_name_id
+  my $table_column = $table_name . "_id";
+  my $id_length = length($id_column_val);
+
+  my $sql = qq/
+    SELECT COUNT(
+      DISTINCT(LEFT($table_column, $id_length))
+    )
+    FROM $table_name
+    WHERE $id_column = $id_column_val
+  /;
+
+  my $results = $helper->execute_single_result(-SQL => $sql);
+  my $test = "$table_column in $table_name is correctly offset by $id_column_val";
```

Comment:
The convention in the other custom test methods (and Perl tests generally) is to accept a `$name` parameter, which overrides anything the test generates. If that doesn't make sense in this case, fair enough, but thought it was worth mentioning.

Reply:
I have applied your suggestions and I think what you have said makes sense in this case too - so have included `$name` parameter and replaced `$test` with it. I will re-request your review to test I have understood :crossed_fingers: